workshops/pixel_art_pad/README.md

```diff
@@ -0,0 +1,462 @@
+    
+    if (!show) noStroke();
+    else stroke(180);
+    
+    rect(x * 20, y * 20, 20, 20);
+  }
+```
+
+Another important point to be noted is that the `Display()` function should be defined inside the Tile class.
+
+#### Specifying the grid size
+
+In order to specify the size of the grid, use the following code :
+
+```java
+Tile grid[][] = new Tile[25][25];
+```
+Here, we are initializing a square grid of 25*25 tiles. You can change it according to your convenience.
+
+#### Specifying the colors in color palette
+
+I love colors. In this workshop, we will be adding 9 colors for our color palette. You can modify or add more colors according to your convenience.
+
+![A variety of colors](https://media.giphy.com/media/8Ry7iAVwKBQpG/giphy.gif)
+
+Use the following code to specify the colors in the color palette :
+
+```java
+color colours[] = {
+  color(165, 42, 42),
+  color(255, 0, 0),
+  color(255, 165, 0),
+  color(255, 255, 0),
+  color(0, 150, 0),
+  color(0, 0, 255),
+  color(255, 0, 200),
+  color(100),
+  color(255)
+};
+
+color selected = color(255);
+boolean show = true;
+```
+
+The colors specified above are in color(R, G, B) format. You can modify them with the R, G, B properties of the color you want.
+
+
+#### Function to setup the canvas
+
+We need to write a function for setting up the canvas. Use the below code to set up the canvas :
+
+```java
+void setup()
+{
+  size(500, 500);
+  
+  for (int a = 0; a < grid.length; ++a)
+  {
+    for (int b = 0; b < grid[a].length; ++b)
+      grid[a][b] = new Tile(a, b, color(100));
+  }
+}
+```
+
+In the above code, the size of the canvas is specified inside `size(500, 500);`. You can modify it according to your need.
+
+
+#### Function to draw upon mouse click
+
+We need to create a function called draw(). This function is used to fill a tile with the seletected color.
+
+```java
+void draw()
+{ 
+  for (int a = 0; a < grid.length; ++a)
+  {
+    for (int b = 0; b < grid[a].length; ++b)
+      grid[b][a].Display();
+  }
+  
+  if (show)
+  {
+    for (int a = 0; a < colours.length; ++a)
+    {
+      fill(colours[a]);
+      stroke(0);
+      ellipse(50, (a * 50) + 50, 30, 30);
+    }
+  }
+  ```
+  
+After defining the draw() function, we should create an 'if' condition to check if the left mouse button is clicked. The following code checks if the left mouse button is pressed and returns the tile which is clicked :
+
+```java
+  if (mousePressed && mouseButton == LEFT)
+  {
+    for (int a = 0; a < grid.length; ++a)
+    {
+      for (int b = 0; b < grid[a].length; ++b)
+      {
+        if (mouseX >= (b * 20) && mouseX <= 20 + (b * 20) && mouseY >= (a * 20) && mouseY <= 20 + (a * 20))
+        {
+          if (show)
+          {
+            for (int c = 0; c < colours.length; ++c)
+            {
+              if (dist(mouseX, mouseY, 50, (c * 50) + 50) <= 30)
+                return;
+            }
+          }
+          
+          grid[b][a].colour = selected;
```

Comment:
these lines aren't really explained much, so it may lead to confusion to the people who do this workshop

Reply:
yes but i thought that the workshop assumes that the person has basic knowledge in java. and since this part is a bit of looping, explaining it would become a bit vague. should I still try to explain it more?

Reply:
yep, u atleast need to explain what we are trying to do with that code and what's happening

Reply:
@hackyguru ^